### ISSUE TITLE
Encountered two children with the same key, `0001-03-01` in the calendar.

### ISSUE THREAD

**Issue by thaynarbo** [rsuite/rsuite#3916]
### What version of rsuite are you using?

5.55.1

### What version of React are you using?

18.2.0

### What version of TypeScript are you using (if any)?

5.4.2

### What browser are you using?

Vitest environment

### Describe the Bug

When I run my unit tests I get this error on the screen

![image](https://github.com/user-attachments/assets/c6331f8f-64a9-45d2-acc3-7283cb06bd2e)

As the screen has no loop that needs to pass a key  I guess it has to do with the calendar component that I added to the screen. 

### Expected Behavior

Calendar component to have unique keys.

**Comment by simonguo** [comment]
Can you provide an example that reproduces the error, preferably by reproducing it in [stackblitz](https://stackblitz.com/edit/react-nxsquq).

**Comment by thaynarbo** [comment]
> Can you provide an example that reproduces the error, preferably by reproducing it in [stackblitz](https://stackblitz.com/edit/react-nxsquq).

Hey there!
Just a heads up I was testing my screen using userSetup and the type method. But the value inside the calendar was very odd. I changed my test to use the selection instead of the typing because the acessibility features were very complicated to test. First I used type(1{arrowRight}12{arrowRight}2024) that for me would be 01/12/2024 as I have it formated like so, but it would go from day to year and at the end all was messed up, by locating the elements was easier and I did not have the error anymore. I'll be closing the issue.
